opencog/nlp/chatbot-psi/chatbot.scm

```diff
@@ -18,51 +18,62 @@
     (cancel-all-threads)
     (reset-all-states)
 
-    (let* ((sent-node (car (nlp-parse utterance)))
-           (list-of-words (get-word-list sent-node)))
-
-        (State input-utterance
-            (Reference
-                list-of-words
-                sent-node
-                (Node utterance)
-            )
+    (catch #t
```

Comment:
ok I'll add comments to the code, the reason is because nlp-parse might fail to parse a sentence, this rarely happens (and should be fixed instead) but when it happens it would be nice to catch the error so that the robot could react accordingly, e.g. look confused or say something like "I don't understand" etc, soon this should be expanded to also detect inputs that are not really grammatical